.github/workflows/build-and-test.yml

```
@@ -81,53 +44,26 @@ jobs:
           # We need to fetch all branches and commits so that Nx affected has a base to compare against.
           fetch-depth: 0
 
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         name: Checkout [Default Branch]
         if: ${{ github.event_name != 'pull_request' }}
         with:
           # We need to fetch all branches and commits so that Nx affected has a base to compare against.
           fetch-depth: 0
 
-      - name: Set up dependencies
-        uses: ./.github/actions/setup
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v4
 
       - name: Initialize the Nx Cloud distributed CI run
-        run: yarn nx-cloud start-ci-run
+        run: npx nx-cloud start-ci-run --distribute-on="./nx/workflows/dynamic-changesets.yml" --stop-agents-after="e2e"
+
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
 
       - name: Run commands in parallel
         run: |
-          pids=()
-          # list of commands to be run on main has env flag NX_CLOUD_DISTRIBUTED_EXECUTION set to false
-          NX_CLOUD_DISTRIBUTED_EXECUTION=false yarn nx-cloud record -- yarn nx format:check & pids+=($!)
-
-          # list of commands to be run on agents
-          yarn nx affected -t lint --parallel=3 &
-          pids+=($!)
-
-          yarn nx affected -t test --parallel=3 &
-          pids+=($!)
-
-          yarn nx affected -t build --exclude=docs --parallel=3 &
-          pids+=($!)
-
-          yarn nx affected -t e2e --parallel=1 &
-          pids+=($!)
-
-          yarn nx affected -t component-test --parallel=1 &
-          pids+=($!)
-
-          # run all commands in parallel and bail if one of them fails
-          for pid in ${pids[*]}; do
-            if ! wait $pid; then
-              exit 1
-            fi
-          done
-
-          exit 0
-
-      - name: Stop Nx Cloud DTE agents
-        if: ${{ always() }}
-        run: yarn nx-cloud stop-all-agents
+          npx nx-cloud record -- npx nx format:check
+          npx nx affected -t lint build test component-test e2e --parallel=4 --exclude=docs
```

suggestion: Try using the `e2e-ci` tasks. [Configuration docs for the Nx Cypress plugin](https://nx.dev/nx-api/cypress#nxcypress-configuration).

Specify `e2e-ci` both for `nx cloud start-ci-run --stop-agents-after` and `nx affected -t`.

```suggestion
          npx nx affected -t lint build test component-test e2e-ci --parallel=4 --exclude=docs
```

---

@LayZeeDK tried, didn't work. I don't want to mess around with it any longer :D if you have any other suggestion, i'm going to revert the e2e-ci changes

---

@eneajaho

Have you tried ISR with Cypress and Nx Atomizer (`e2e-ci` tasks)?

---

Oh, it's also SSR without ISR

https://cloud.nx.app/runs/16A7H6itJx/task/ssr%3Ae2e-ci--cypress%2Fe2e%2Fapp.cy.ts